engine/src/main/java/org/hibernate/validator/internal/engine/constraintvalidation/ConstraintValidatorManager.java

```diff
@@ -223,11 +233,22 @@ public int numberOfCachedConstraintValidatorInstances() {
 		return determinedSuitableTypes;
 	}
 
-	private <A extends Annotation> void initializeValidator(ConstraintDescriptor<A> descriptor, ConstraintValidator<A, ?> constraintValidator) {
+	private <A extends Annotation> void initializeValidator(
+			ConstraintDescriptor<A> descriptor,
+			ConstraintValidator<A, ?> constraintValidator,
+			HibernateConstraintValidatorInitializationContext initializationContext) {
 		try {
-			constraintValidator.initialize( descriptor.getAnnotation() );
+			if ( constraintValidator instanceof HibernateConstraintValidator ) {
+				( (HibernateConstraintValidator) constraintValidator ).initialize( descriptor, initializationContext );
+			}
+			else {
+				constraintValidator.initialize( descriptor.getAnnotation() );
+			}
 		}
 		catch (RuntimeException e) {
+			if ( e instanceof ConstraintDeclarationException ) {
+				throw e;
+			}
```

Comment:
@marko-bekhta I just noticed this change. What does it have to do with this patch?

Reply:
it's from those script assert constraints. In `ScriptAssertValidatorTest#unknownLanguageNameRaisesException()`  `ConstraintDeclarationException` is expected when no language can be found. If run that test without this change we'd get:
```
javax.validation.ValidationException: HV000032: Unable to initialize org.hibernate.validator.internal.constraintvalidators.hv.ScriptAssertValidator.
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintValidatorManager.initializeValidator(ConstraintValidatorManager.java:258)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintValidatorManager.createAndInitializeValidator(ConstraintValidatorManager.java:170)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintValidatorManager.getInitializedValidator(ConstraintValidatorManager.java:126)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.getInitializedConstraintValidator(ConstraintTree.java:208)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.getInitializedConstraintValidator(ConstraintTree.java:185)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateConstraints(ConstraintTree.java:115)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateConstraints(ConstraintTree.java:86)
	at org.hibernate.validator.internal.metadata.core.MetaConstraint.doValidateConstraint(MetaConstraint.java:116)
	at org.hibernate.validator.internal.metadata.core.MetaConstraint.validateConstraint(MetaConstraint.java:109)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateMetaConstraint(ValidatorImpl.java:570)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForSingleDefaultGroupElement(ValidatorImpl.java:528)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForDefaultGroup(ValidatorImpl.java:497)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForCurrentGroup(ValidatorImpl.java:462)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateInContext(ValidatorImpl.java:412)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validate(ValidatorImpl.java:200)
	at org.hibernate.validator.test.internal.constraintvalidators.hv.ScriptAssertValidatorTest.unknownLanguageNameRaisesException(ScriptAssertValidatorTest.java:111)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:80)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
	... 17 more
Caused by: javax.validation.ConstraintDeclarationException: HV000170: No JSR-223 scripting engine could be bootstrapped for language "foo".
	at org.hibernate.validator.internal.constraintvalidators.hv.AbstractScriptAssertValidator.initializeScriptContext(AbstractScriptAssertValidator.java:37)
	at org.hibernate.validator.internal.constraintvalidators.hv.ScriptAssertValidator.initialize(ScriptAssertValidator.java:48)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintValidatorManager.initializeValidator(ConstraintValidatorManager.java:248)
	... 38 more
Caused by: org.hibernate.validator.spi.scripting.ScriptEvaluatorNotFoundException: HV000232: No JSR 223 script engine found for language "foo".
	at org.hibernate.validator.internal.engine.scripting.DefaultScriptEvaluatorFactory.createNewScriptEvaluator(DefaultScriptEvaluatorFactory.java:66)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.hibernate.validator.spi.scripting.AbstractCachingScriptEvaluatorFactory.getScriptEvaluatorByLanguageName(AbstractCachingScriptEvaluatorFactory.java:41)
	at org.hibernate.validator.internal.engine.constraintvalidation.HibernateConstraintValidatorInitializationContextImpl.getScriptEvaluatorForLanguage(HibernateConstraintValidatorInitializationContextImpl.java:43)
	at org.hibernate.validator.internal.constraintvalidators.hv.AbstractScriptAssertValidator.initializeScriptContext(AbstractScriptAssertValidator.java:33)
	... 40 more
```

And personally I'd be OK with such error as the cause is clear, but I didn't want to change the behavior of which exception is thrown (maybe somebody is depending on it ?) WDYT?

Reply:
OK, makes sense.